Add ?raw to support requests for raw package source files

johnyanarella opened the pull request:
As a followup to a Discord discussion, this PR adds an escape hatch via a new `?raw` query (and extra query) to allow requests for raw (untransformed) package source files.

In rare cases, you may want to request JS source files from packages, as-is, without transformation into ES modules. To do so, you would now add a `?raw` query to the request URL.

For example, you might need to register a package's source script as a service worker in a browser that [does not yet support](https://caniuse.com/mdn-api_serviceworker_ecmascript_modules) the `type: "module"` option:

```js
await navigator.serviceWorker.register(
  new URL(
    "https://esm.sh/playground-elements@0.18.1/playground-service-worker.js?raw", 
    import.meta.url.href
  ), 
  { scope: '/' }
);
```

You could also now alternatively specify an `&raw` extra query after the package version:

```html
<playground-project sandbox-base-url="https://esm.sh/playground-elements@0.18.1&raw/"
></playground-project>
```

so that transitive references in raw assets will also be raw requests.

(i.e. where an existing HTML package asset you request needs to subsequently request and receive a raw .js package asset)

As a concrete example benefit of this change, this new feature enables ESM.sh (or a self-hosted instance of ESM.sh with private nom repository) to be used like `unpkg.com` (the default) as the sandbox base URL for [Google Playground Elements](https://github.com/google/playground-elements).

ije commented on server/server_handler.go:
```diff
@@ -725,6 +725,8 @@ func esmHandler() rex.Handle {
 					}
 					reqType = "types"
+				} else if ctx.R.URL.Query().Has("raw") {
+					reqType = "raw"
```
nice, i didn't realize how easy to add this feature!

ije commented:
@johnyanarella can you please also add checking of the `?raw` for `esm-worker`?
https://github.com/esm-dev/esm.sh/blob/a8972d1320f5cbac22c94130738e7eccc4f0dda4/packages/esm-worker/src/index.ts#L650

johnyanarella commented:
Updated to add `?raw` (and `&raw`) support in esm-worker (and a corresponding test).

And hopefully also fixed the inconsistency for the `Content-Type` value across platforms (i.e. the different resulting content type for JS between local dev and CI that occurred in that Testing workflow run).

ije reviewed (approved):
LGTM, thanks 🙏
